File: app/src/main/java/com/mapmyindia/sdk/demo/java/activity/GesturesActivity.java
package com.mapmyindia.sdk.demo.java.activity;

import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.mapbox.android.gestures.AndroidGesturesManager;
import com.mapbox.android.gestures.MoveGestureDetector;
import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
import com.mapbox.mapboxsdk.geometry.LatLng;
import com.mapbox.mapboxsdk.maps.MapView;
import com.mapbox.mapboxsdk.maps.MapboxMap;
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
import com.mapmyindia.sdk.demo.R;

public class GesturesActivity extends AppCompatActivity implements OnMapReadyCallback {

    private MapView mapView;
    private MapboxMap mapmyIndiaMap;
    private AndroidGesturesManager androidGesturesManager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.base_layout);
        mapView = findViewById(R.id.map_view);
        mapView.onCreate(savedInstanceState);
        mapView.getMapAsync(this);
        androidGesturesManager = new AndroidGesturesManager(mapView.getContext(), false);
        addTouchGesture();
    }

    private void addTouchGesture() {

        if(androidGesturesManager != null) {


            mapView.setOnTouchListener(new View.OnTouchListener() {
                @Override
                public boolean onTouch(View v, MotionEvent event) {

                    androidGesturesManager.onTouchEvent(event);
                    return false;
                }


            });
            androidGesturesManager.setMoveGestureListener(new MoveGestureDetector.OnMoveGestureListener() {
                @Override
                public boolean onMoveBegin(MoveGestureDetector detector) {

                    if (mapmyIndiaMap != null && detector.getPointersCount() == 1) {
                        LatLng latLng = mapmyIndiaMap.getProjection().fromScreenLocation(detector.getFocalPoint());
                        Toast.makeText(GesturesActivity.this, "onMoveBegin: " + latLng.toString(), Toast.LENGTH_SHORT).show();

                    }

                    return true;
                }

                @Override
                public boolean onMove(MoveGestureDetector detector, float distanceX, float distanceY) {
                    if (mapmyIndiaMap != null && detector.getPointersCount() == 1) {
                        LatLng latLng = mapmyIndiaMap.getProjection().fromScreenLocation(detector.getFocalPoint());
                        Toast.makeText(GesturesActivity.this, "onMove: " + latLng.toString(), Toast.LENGTH_SHORT).show();


                    }
                    return false;
                }

                @Override
                public void onMoveEnd(MoveGestureDetector detector, float velocityX, float velocityY) {
                    if (mapmyIndiaMap != null) {
                        LatLng latLng = mapmyIndiaMap.getProjection().fromScreenLocation(detector.getFocalPoint());
                        Toast.makeText(GesturesActivity.this, "onMoveEnd: " + latLng.toString(), Toast.LENGTH_SHORT).show();

                    }
                }
            });
        }
    }

    @Override
    public void onMapReady(MapboxMap mapmyIndiaMap) {
        this.mapmyIndiaMap = mapmyIndiaMap;

        mapmyIndiaMap.animateCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(28, 77), 14));


    }

    @Override
    public void onMapError(int i, String s) {
    }

    @Override
    protected void onStart() {
        super.onStart();
        mapView.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
        mapView.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mapView.onDestroy();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mapView.onPause();
    }

    @Override
    protected void onResume() {
        super.onResume();
        mapView.onResume();
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
        mapView.onLowMemory();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        mapView.onSaveInstanceState(outState);
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/activity/GesturesActivity.kt
package com.mapmyindia.sdk.demo.kotlin.activity

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.mapmyindia.sdk.demo.R
import com.mapmyindia.sdk.gestures.AndroidGesturesManager
import com.mapmyindia.sdk.gestures.MoveGestureDetector
import com.mapmyindia.sdk.maps.MapView
import com.mapmyindia.sdk.maps.MapmyIndiaMap
import com.mapmyindia.sdk.maps.OnMapReadyCallback
import com.mapmyindia.sdk.maps.camera.CameraUpdateFactory
import com.mapmyindia.sdk.maps.geometry.LatLng

class GesturesActivity : AppCompatActivity(), OnMapReadyCallback {
    private lateinit var mapView: MapView
    private var mapmyIndiaMap: MapmyIndiaMap? = null
    private var androidGesturesManager: AndroidGesturesManager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.base_layout)
        mapView = findViewById<MapView>(R.id.map_view) as MapView
        mapView.onCreate(savedInstanceState)
        mapView.getMapAsync(this)
        androidGesturesManager = AndroidGesturesManager(mapView.context, false)
        addTouchGesture()
    }

    private fun addTouchGesture() {

        mapView.setOnTouchListener { v, event ->
            androidGesturesManager?.onTouchEvent(event)
            return@setOnTouchListener false
        }

        androidGesturesManager?.setMoveGestureListener(object : MoveGestureDetector.OnMoveGestureListener {
            override fun onMoveBegin(detector: MoveGestureDetector): Boolean {
                if (mapmyIndiaMap != null && detector.pointersCount == 1) {
                    val latLng = mapmyIndiaMap?.projection?.fromScreenLocation(detector.focalPoint)
                    Toast.makeText(this@GesturesActivity, "onMoveBegin: $latLng", Toast.LENGTH_SHORT).show()
                }

                return true
            }

            override fun onMove(detector: MoveGestureDetector, distanceX: Float, distanceY: Float): Boolean {
                if (mapmyIndiaMap != null && detector.pointersCount == 1) {
                    val latLng = mapmyIndiaMap?.projection?.fromScreenLocation(detector.focalPoint)
                    Toast.makeText(this@GesturesActivity, "onMove: $latLng", Toast.LENGTH_SHORT).show()
                }
                return false
            }

            override fun onMoveEnd(detector: MoveGestureDetector, velocityX: Float, velocityY: Float) {
                if (mapmyIndiaMap != null) {
                    val latLng = mapmyIndiaMap?.projection?.fromScreenLocation(detector.focalPoint)
                    Toast.makeText(this@GesturesActivity, "onMoveEnd: $latLng", Toast.LENGTH_SHORT).show()
                }
            }

        })
    }

    override fun onMapError(p0: Int, p1: String?) {}

    override fun onMapReady(mapmyIndiaMap: MapmyIndiaMap) {
        this.mapmyIndiaMap = mapmyIndiaMap;


        mapmyIndiaMap.animateCamera(CameraUpdateFactory.newLatLngZoom(LatLng(28.0, 77.0), 14.0))
    }

    override fun onStart() {
        super.onStart()
        mapView.onStart()
    }

    override fun onResume() {
        super.onResume()
        mapView.onResume()
    }

    override fun onPause() {
        super.onPause()
        mapView.onPause()
    }

    override fun onStop() {
        super.onStop()
        mapView.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
        mapView.onDestroy()
    }

    override fun onLowMemory() {
        super.onLowMemory()
        mapView.onLowMemory()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        mapView.onSaveInstanceState(outState)
    }
}
